File: Array/linchuanxin/Algorithm/day47_0717_相交链表/day47_leetcode160.java
package day47_0717;

import java.util.List;

public class day47_leetcode160 {


      public class ListNode {
      int val;
      ListNode next;
      ListNode(int x) {
          val = x;
          next = null;
      }
  }

    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
          ListNode p = headA;
          ListNode q = headB;

          while (p!=q){
              if (p!=null){
                  p = p.next;
              }else{
                  p = headB;
              }
              if(q!=null){
                  q = q.next;
              }else{
                  q  = headA;
              }
          }
          return p;
    }
}

File: Array/linchuanxin/Algorithm/day47_0717_相交链表/day47_leetcode160.kt
package day47_0717


class ListNode(var `val`: Int) {
         var next: ListNode? = null
     }


fun getIntersectionNode(headA:ListNode?, headB:ListNode?):ListNode? {
    var p: ListNode? = headA
    var q:ListNode? = headB

    while (p != q){
        if (p != null)
            p = p.next
        else
            p = headB
        if (q != null)
            q = q.next
        else
            q = headA
    }
    return p
}